prediction_market_agent/agents/microchain_agent/deploy.py

```diff
@@ -97,64 +103,99 @@ def run_general_agent(
-                    goal=goal,
-                    chat_history=ChatHistory.from_list_of_dicts(agent.history),
-                )
-                goal_manager.save_evaluated_goal(
-                    goal=goal,
-                    evaluation=goal_evaluation,
+        iteration = 0
+        while self.max_iterations is None or iteration < self.max_iterations:
+            starting_history_length = len(agent.history)
+            try:
+                # After the first iteration, resume=True to not re-initialize the agent.
+                agent.run(iterations=1, resume=iteration > 0)
+            except Exception as e:
+                logger.error(f"Error while running microchain agent: {e}")
+                raise e
+            finally:
+                # Save the agent's history to the long-term memory after every iteration to keep users updated.
+                save_agent_history(
+                    agent=agent,
+                    long_term_memory=long_term_memory,
+                    initial_system_prompt=initial_formatted_system_prompt,
+                    # Because the agent is running in a while cycle, always save into database only what's new, to not duplicate entries.
+                    save_last_n=len(agent.history) - starting_history_length,
                 )
-                agent.history.append(
-                    ChatMessage(
-                        role="user",
-                        content=str(f"# Goal evaluation\n{goal_evaluation}"),
-                    ).model_dump()
-                )
-
+                if agent.system_prompt != initial_formatted_system_prompt:
+                    prompt_handler.save_prompt(get_editable_prompt_from_agent(agent))
+            iteration += 1
+
+        if goal_manager:
+            goal = check_not_none(goal)
+            goal_evaluation = goal_manager.evaluate_goal_progress(
+                goal=goal,
+                chat_history=ChatHistory.from_list_of_dicts(agent.history),
+            )
+            goal_manager.save_evaluated_goal(
+                goal=goal,
+                evaluation=goal_evaluation,
+            )
+            agent.history.append(
+                ChatMessage(
+                    role="user",
+                    content=f"# Goal evaluation\n{goal_evaluation}",
+                ).model_dump()
+            )
             save_agent_history(
                 agent=agent,
                 long_term_memory=long_term_memory,
                 initial_system_prompt=initial_formatted_system_prompt,
+                save_last_n=1,
             )
-            if agent.system_prompt != initial_formatted_system_prompt:
-                prompt_handler.save_prompt(get_editable_prompt_from_agent(agent))
+
+
+class DeployableMicrochainAgent(DeployableMicrochainAgentAbstract):
+    identifier = AgentIdentifier.MICROCHAIN_AGENT_OMEN
+    functions_config = TRADING_AGENT_SYSTEM_PROMPT_CONFIG.functions_config
```

Comment:
Because system prompts are now part of the deployable agent class for general agents, we need to specify functions config as well. 

For the pre-defined prompts, I created base class that holds the system prompt and functions config, so they can remain linked.